### ISSUE TITLE
Fix maven integration tests to work with security manager

### ISSUE THREAD

**Issue by rjernst** [elastic/elasticsearch#12218]
Maven integration tests currently do not work with security manager. There are a couple easy to fix issues like passing the right system properties to give access to .class files. However, a larger issue is the forked classloader the maven failsafe plugin uses. We can disable this isolated classloader with `useManifestOnlyJar=false`. However, the jars from maven itself then have jar hell (the failsafe api jar and failsafe junit jar have duplicate classes). We might be able to exclude the api jar (assuming the junit jar is a superset of the classes).

**Comment by rjernst** [comment]
I got a little farther in removing jar hell:
https://gist.github.com/rjernst/32f1666b8a6fd955e085

However, this surefire/failsafe plugin is just messy. This is the current failure:
```
Caused by: java.lang.IllegalStateException: jar hell!
class: org.apache.maven.shared.utils.StringUtils
jar1: /Users/rjernst/.m2/repository/org/apache/maven/surefire/common-java5/2.18.1/common-java5-2.18.1.jar
jar2: /Users/rjernst/.m2/repository/org/apache/maven/shared/maven-shared-utils/0.4/maven-shared-utils-0.4.jar
```

I'm not sure it is worthwhile to fix this. Security manager is already running for the actual ES server for the tests. This is only for rest test itself (ie the client). One thing to try might be using junit4 as we do for unit tests.

**Comment by s1monw** [comment]
> I'm not sure it is worthwhile to fix this. Security manager is already running for the actual ES server for the tests. This is only for rest test itself (ie the client). One thing to try might be using junit4 as we do for unit tests.

I agree - the actual service that we are testing (the unzipped elasticsearch server) starts up with sec manager enabled so I think we should be just fine here?

**Comment by rmuir** [comment]
By the way, this is fixed with #12302

Although its true in this case, its unimportant for testing (since the external cluster runs with security manager and thats the important part), i feel better with integ tests not running unrestrained.
